chef/automate
# PG changes release version 3.x

### :nut_and_bolt: Description: What code changed, and why?


This is to upgrade the postgresql version to 13 and addition of migration command.  
This PR contains all the changes that were earlier merged into this base branch.
Also fixed notifications service failure to send HTTPS requests
This also contains all the changes required to integration pipelines to solve issues occurring due to semver changes and pg changes

### :chains: Related Resources

https://chefio.atlassian.net/browse/SHIELD-22?atlOrigin=eyJpIjoiMGY1ODg2OTQ2Y2IyNDBiNzljOTc3NWM0YzgwMTdkMzciLCJwIjoiaiJ9

### :+1: Definition of Done

Pipelines should be green. 
Upgrading to this new version should have postgresql13 installed. 
Following commands should be available and working without error:
- `chef-automate post-major-upgrade migrate --data=PG`
- `chef-automate post-major-upgrade clear-data --data=PG`
- All services should be up and running
- If migrating second time it should allow with yes or no.
- If data is not migrated it should not let further upgrade
- Use all the flags for example: skip-storage-check, --force

### :athletic_shoe: How to Build and Test the Change

- Result/build.json should have milestone manifest 20220329091442
- `hab studio enter`
- `start_all_services`
- Add some dummy data
- Upgrade automate to 3.0.6 by running `chef-automate upgrade run --major`
- `rebuild components/automate-postgresql`
- `rebuild components/pg-sidecar-service`
- `rebuild components/automate-cli`
- `rebuild components/notifications-service`
- Run the migrate command `chef-automate post-major-upgrade migrate --data=PG`
- Run the clear data command `chef-automate post-major-upgrade clear-data --data=PG`

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
